osf/models/sanctions.py

```diff
@@ -167,6 +186,23 @@ def _on_complete(self, user):
     def forcibly_reject(self):
         self.state = Sanction.REJECTED
 
+    def _save_transition(self, event_data):
+        """Recored the effects of a state transition in the database."""
+        self.save()
+        new_state = event_data.transition.dest
+        # No need to update registration state with no sanction state change
+        if new_state is None:
+            return
+
+        user = event_data.kwargs.get('user')
+        if user is None and event_data.kwargs:
+            user = event_data.args[0]
+        comment = event_data.kwargs.get('comment', '')
+        if new_state == ApprovalStates.PENDING_MODERATION.name:
+            user = None  # Don't worry about the particular user who gave final approval
```

Comment:
hm, i feel like this function knows/assumes too much about the states and transitions we happen to use at the moment, which seems brittle/smelly... but probably within osf norms (edit: and isn't new in this pr), so just a nitpick

Reply:
Agreed. This was triggering specific logic that the Registration should be explicitly confirming on its own. Moved the check there